tests/data/short_circuit/single_phase_to_ground_a/input.json

```diff
@@ -26,6 +26,6 @@
     ],
   "fault":
     [
-      {"id": 10, "r": 0.1, "x": 0.1, "node": 3, "status": 1}
+      {"id": 10, "r": 0.1, "x": 0.1, "fault_type": "single_phase_to_ground", "phase": "a", "fault_object": 3, "status": 1}
```

Comment:
I don't know how to solve these fault types in the input/update type.

My proposal is either have a separate field for the `fault_type` and the phases like so:

```json
        {
            "name": "FaultInput",
            "base": "BaseInput",
            "attributes": [
                {
                    "data_type": "ID",
                    "names": "fault_object",
                    "description": "ID of the faulty object"
                },
                {
                    "data_type": "IntS",
                    "names": "status",
                    "description": "Whether the fault is active (object is faulty)"
                },
                {
                    "data_type": "FaultType",
                    "names": "fault_type",
                    "description": "type of fault"
                },
                {
                    "data_type": "FaultPhases",
                    "names": "fault_phases",
                    "description": "phases of fault"
                },
                {
                    "data_type": "double",
                    "names": [
                        "r_f",
                        "x_f"
                    ],
                    "description": "short circuit impedance"
                }
            ]
        }
```

or to have separate fault types for each phase set:

```json
        {
            "name": "FaultInput",
            "base": "BaseInput",
            "attributes": [
                {
                    "data_type": "ID",
                    "names": "fault_object",
                    "description": "ID of the faulty object"
                },
                {
                    "data_type": "IntS",
                    "names": "status",
                    "description": "Whether the fault is active (object is faulty)"
                },
                {
                    "data_type": "FaultType",
                    "names": "fault_type",
                    "description": "type of fault"
                },
                {
                    "data_type": "double",
                    "names": [
                        "r_f",
                        "x_f"
                    ],
                    "description": "short circuit impedance"
                }
            ]
        }
```

Reply:
In https://github.com/PowerGridModel/power-grid-model/pull/233 it was decided to remove this from the component and into the calculation options interface. To be honest, I don't know whether that is the correct option. I'm thinking about compatibility here. I'm considering the following use cases:

* batch calculation for different fault types on the same node
  * requires to have it available in the update data
* if the case of having multiple faults is considered (cfr. the reason as to why the sequential short circuit exists mentioned in the chapter introduction in https://www.phasetophase.nl/boek/boek_2_10.html#_10 ), then i want to add multiple faults, possibly with different fault types.
  * In those cases, i guess one wants to treat multiple faults sequentially (e.g. by first solving for the first, then feeding that result as input to the second fault, etc., rather than by constructing a single optimization function and then solving that).
  * To avoid exposing this complexity to the user, we should just allow a list of faults to be provided. How we handle that is a `calculation_method` choice and should not require multiple API calls to `calculate` by the user

Reply:
cfr. discussions: this was accepted and resolved